tensorflow/core/kernels/segment_reduction_ops_impl.h

```diff
@@ -110,7 +110,7 @@ class SegmentReductionOp : public OpKernel {
                 errors::InvalidArgument("Shape must be at least rank 1"));
 
     TensorShape output_shape = input.shape();
-    output_shape.set_dim(0, output_rows);
+    OP_REQUIRES_OK(context, output_shape.SetDimWithStatus(0, output_rows));
```

Comment:
Why doesn't the 

```
    OP_REQUIRES(context, output_rows >= 0,
                errors::InvalidArgument("segment ids must be >= 0"));
```

check above catch this?

Reply:
`SetDim` also checks that the total tensor size does not overflow.

Reply:
Thanks, maybe add a comment?